File: ported/auto/40a579.js
// Auto-translated from Ghidra C by tools/c-to-js/translate.js.
// Source: decompiled/c/40a579.c
// Edit by hand only after diff-test passes — re-running the translator will overwrite.

/** @typedef {import("../../runtime/heap.js").Heap} Heap */

export function FUN_0040a579(heap, param_1, param_2) {
  let uVar1 = 0;
  if ((heap.u32(0x005ebf5c) == 0) && (heap.u32(0x005ebf60) == 0)) {
    if ((heap.u32((param_1 + 0x80)) == 0) || (heap.u32((param_2 + 0x80)) == 0)) {
      uVar1 = 0;
    } else {
      heap.setU32(0x005ebf5c, (heap.u32((param_1 + 0x80))) >>> 0);
      heap.setU32(0x005ebf60, (heap.u32((param_2 + 0x80))) >>> 0);
      uVar1 = 1;
    }
  } else {
    uVar1 = 0;
  }
  return uVar1;
}

File: decompiled/c/40a579.c

undefined4 FUN_0040a579(int param_1,int param_2)

{
  undefined4 uVar1;
  
  if ((DAT_005ebf5c == 0) && (DAT_005ebf60 == 0)) {
    if ((*(int *)(param_1 + 0x80) == 0) || (*(int *)(param_2 + 0x80) == 0)) {
      uVar1 = 0;
    }
    else {
      DAT_005ebf5c = *(int *)(param_1 + 0x80);
      DAT_005ebf60 = *(int *)(param_2 + 0x80);
      uVar1 = 1;
    }
  }
  else {
    uVar1 = 0;
  }
  return uVar1;
}

